src/test/java/teammates/test/cases/ui/StudentProfilePictureActionTest.java

```diff
@@ -189,22 +190,85 @@ private void testActionForStudentWithEmptyGoogleId() {
         assertEquals("", result.blobKey);
     }
 
-    protected void testActionWithEmailAndCourseUnauthorisedInstructor() {
+    protected void testActionWithEmailAndCourseUnauthorisedInstructorOrStudent() {
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.STUDENT_EMAIL, StringHelper.encrypt(student.email),
+                Const.ParamsNames.COURSE_ID, StringHelper.encrypt(student.course)
+        };
+        
         ______TS("Failure case: instructor not from same course");
         AccountAttributes unauthInstructor = dataBundle.accounts.get("instructor1OfCourse2");
         gaeSimulation.loginAsInstructor(unauthInstructor.googleId);
 
+        action = getAction(submissionParams);
+        try {
+            action.executeAndPostProcess();
+            signalFailureToDetectException("Unauthorised Access");
+        } catch (UnauthorizedAccessException uae) {
+            assertEquals("User is not in the course that student belongs to", uae.getMessage());
+        }
```

Comment:
In that case, is it too much work to modify one instructor and remove its "viewing student" privilege? I am not entirely sure on the details, but this should be doable via some BackDoor methods. Do not change the typical data bundle since it will have widespread effect.